runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -101,9 +102,8 @@ public void copyFrom(ParserRuleContext ctx) {
 			this.children = new ArrayList<>();
 			// reset parent pointer for any error nodes
 			for (ParseTree child : ctx.children) {
-				if ( child instanceof ErrorNodeImpl ) {
-					this.children.add(child);
-					((ErrorNodeImpl) child).parent = this;
+				if ( child instanceof ErrorNode ) {
+					addChild((ErrorNode)child);
```

Comment:
As I said, a bit weird. You have a copy function which moves something :-) At least I have it then like you want. But the Java code is still buggy. The error node is still in the children list of the source context, since you don't remove it from there.